### ISSUE TITLE
addToDocument not working in safari

### ISSUE THREAD

**Issue by chandu0101** [japgolly/scalacss#11]
StyleSheet.addToDocument() not working in safari

**Comment by japgolly** [comment]
Internet says this is a workaround for safari:
```
style.appendChild(document.createTextNode("...")); // for webkit.
```

I don't have Safari, can you try adding the above hack to [React.scala#L32-L35](https://github.com/japgolly/scalacss/blob/master/ext-react/src/main/scala/japgolly/scalacss/React.scala#L32-L35) and see if it works for you?

**Comment by chandu0101** [comment]
no luck :( , dam you DOM ...

**Comment by japgolly** [comment]
Really? Damn :( There has to be a way. We'll find it.

Can you paste the snippet that you tried that didn't work?
​

**Comment by chandu0101** [comment]
i added empty text before adding our css 

```scala
class StyleElementRenderer(s: Renderer[String]) extends Renderer[HTMLStyleElement] {
    override def apply(css: Css) = {
      val e = document.createElement("style").asInstanceOf[HTMLStyleElement]
      e.`type` = "text/css"
      e appendChild document.createTextNode("")
      e appendChild document.createTextNode(s(css))
      e
    }
  }
```

**Comment by japgolly** [comment]
What about this?
```scala
  class StyleElementRenderer(s: Renderer[String]) extends Renderer[HTMLStyleElement] {
    override def apply(css: Css) = {
      val content = s(css)
      val e = document.createElement("style").asInstanceOf[HTMLStyleElement]
      e.`type` = "text/css"
      e.setAttribute("rel", "stylesheet")
      e.style.cssText = content
      e appendChild document.createTextNode(content)
      e
    }
  }
```

**Comment by japgolly** [comment]
Someone else on the internet is saying
```js
if (css.styleSheet) css.styleSheet.cssText = styles;
else css.appendChild(document.createTextNode(styles));
```

**Comment by chandu0101** [comment]
yeah i tried many workarounds mentioned on internet including innnerHTML ,but none worked! .let me try again to see if i missed ...

**Comment by japgolly** [comment]
Maybe instead of using Scala.JS DOM like `e.style.cssText = content` we should try using `js.Dynamic` to do this: `if (css.styleSheet) css.styleSheet.cssText = styles`.

Maybe `.style` is standard but `.styleSheet` is Safari only or something?

**Comment by chandu0101** [comment]
i just tried  ```e.style.cssText = content```` hack didn't worked!

```css.styleSheet```  -> undefined safari web console

**Comment by japgolly** [comment]
To be clear, I mean this:
```scala
val e = document.createElement("style").asInstanceOf[js.Dynamic]
if (e.styleSheet) e.styleSheet.cssText = content
```
That doesn't work either?

**Comment by chandu0101** [comment]
hmm in safari console i tried this 

```
> var css = document.createElement("style")
< undefined
> css.style
< CSSStyleDeclaration

> css.styleSheet
< undefined

```

**Comment by japgolly** [comment]
Damn :( Why is the internet lying to us??!

I don't know mate. Internet people claim it works (in different ways), there must be something we're missing. I don't think I can help much without Safari.

**Comment by ngbinh** [comment]
@japgolly Do you have a kitchen sink demo page somewhere so people can try on different browsers? Doesn't need to include everything!

**Comment by japgolly** [comment]
@ngbinh That's a really cool idea but no, don't have yet.

**Comment by chandu0101** [comment]
it looks like something wrong with my safari :( , i just re installed safari 8 , i see styles in head :)
but prefix is not applied for display:flex

```css
.scalacss-0001 {
display: flex;
-webkit-justify-content: space-between;
justify-content: space-between;
}
```
